compressed_depth_image_transport/src/compressed_depth_publisher.cpp

```diff
@@ -33,57 +33,176 @@
 *********************************************************************/
 
 #include "compressed_depth_image_transport/compressed_depth_publisher.h"
-#include <cv_bridge/cv_bridge.hpp>
-#include <sensor_msgs/image_encodings.hpp>
-#include <opencv2/highgui/highgui.hpp>
-
-#include "compressed_depth_image_transport/codec.h"
 #include "compressed_depth_image_transport/compression_common.h"
+#include "compressed_depth_image_transport/codec.h"
 
 #include <rclcpp/parameter_client.hpp>
+#include <rclcpp/parameter_events_filter.hpp>
 
-#include <vector>
-#include <sstream>
-
-constexpr int kDefaultPngLevel = 9;
-constexpr double kDefaultDepthMax = 10.0;
-constexpr double KDefaultDepthQuantization = 100.0;
-
-using namespace cv;
-using namespace std;
+namespace compressed_depth_image_transport
+{
 
-namespace enc = sensor_msgs::image_encodings;
+enum compressedDepthParameters
+{
+  DEPTH_MAX = 0,
+  DEPTH_QUANTIZATION,
+  PNG_LEVEL
+};
 
-namespace compressed_depth_image_transport
+const struct ParameterDefinition kParameters[] =
 {
+  { //DEPTH_MAX - Maximum depth value (meter)
+    ParameterValue((double)10.0),
+    ParameterDescriptor()
+      .set__name("depth_max")
+      .set__type(rcl_interfaces::msg::ParameterType::PARAMETER_DOUBLE)
+      .set__description("Maximum depth value (meter)")
+      .set__read_only(false)
+      .set__floating_point_range(
+        {rcl_interfaces::msg::FloatingPointRange()
+          .set__from_value(1.0)
+          .set__to_value(100.0)
+          .set__step(0.0)})
+  },
+  { //DEPTH_QUANTIZATION - Depth value at which the sensor accuracy is 1 m (Kinect: >75)
+    ParameterValue((double)100.0),
+    ParameterDescriptor()
+      .set__name("depth_quantization")
+      .set__type(rcl_interfaces::msg::ParameterType::PARAMETER_DOUBLE)
+      .set__description("Depth value at which the sensor accuracy is 1 m (Kinect: >75)")
+      .set__read_only(false)
+      .set__floating_point_range(
+        {rcl_interfaces::msg::FloatingPointRange()
+          .set__from_value(1.0)
+          .set__to_value(150.0)
+          .set__step(0.0)})
+  },
+  { //PNG_LEVEL - PNG Compression Level from 0 to 9.  A higher value means a smaller size.
+    ParameterValue((int)3), //Default to OpenCV default of 3
```

Comment:
Let's keep this as 9 to not change behavior

```suggestion
    ParameterValue((int)9),
```

Reply:
Ok, we can do it.

What may happen:
- both `compressed` and `compressedDepth` will declare `<some_path>.<image>.png_level`
- they will "race" on setting the default parameter value for the node
- one of them wins and sets the default for deprecated parameter
- actually I am not sure if we end up with the first or second value

Looking at 

```bash
ros2 run image_transport list_transports 
Declared transports:
image_transport/compressed
image_transport/compressedDepth
image_transport/raw
image_transport/theora
```

`compressed` seems to be the first to set its value.

This is decided here:
- https://github.com/ros-perception/image_common/blob/843443f6fa57f8ad00a7b0ffec0dbad796917ed1/image_transport/src/publisher.cpp#L120
- it's not really a race but depends on plugin order load

-------------------

This is what we try to avoid for the future in #140 and why we need to remove non-scoped parameters.

Reply:
I am not sure yet what to do with this one.

I don't like setting `png_level` 9 as it is non real-time value on most machines.

This doesn't work at 30 fps in most cases, limits publishing frequency and blocks some nodes partially (in ROS1, my experience with ROS2 is limited here)
- https://github.com/IntelRealSense/realsense-ros/issues/1672
- https://github.com/stereolabs/zed-ros-wrapper/issues/875

I would not care much if it is only for `compressedDepth` (and we keep it as it was)
but the "race" makes we wonder if `compressed` will also not get it.

Reply:
Given
* compressed seems to be the first to set its value
and assuming:
* most people who use compressed depth image transport would also have compressed depth transport installed.
I think most people would rely currently on the default png_level being being 3 for the compressed depth transport. If that's the case, I'm fine with having 3 to not break behavior.